content/knowledge-base/guides/write-a-composition-function-in-go.md

```diff
@@ -0,0 +1,826 @@
+  generateName: example-buckets-
+  labels:
+    crossplane.io/composite: example-buckets
+  ownerReferences:
+    # Omitted for brevity
+spec:
+  forProvider:
+    region: us-east-2
+```
+
+{{<hint "tip">}}
+Read the composition functions documentation to learn more about
+[testing composition functions]({{<ref "../../master/concepts/composition-functions#test-a-composition-that-uses-functions">}}).
+{{</hint>}}
+
+## Build and push the function to a package registry
+
+You build a function in two stages. First you build the function's runtime. This
+is the Open Container Initiative (OCI) image Crossplane uses to run your
+function. You then embed that runtime in a package, and push it to a package
+registry. The Crossplane CLI uses `xpkg.upbound.io` as its default package
+registry.
+
+A function supports a single platform, like `linux/amd64`, by default. You can
+support multiple platforms by building a runtime and package for each platform,
+then pushing all the packages to a single tag in the registry.
+
+Pushing your function to a registry allows you to use your function in a
+Crossplane control plane. See the
+[composition functions documentation]({{<ref "../../master/concepts/composition-functions">}}).
+to learn how to use a function in a control plane.
+
+Use Docker to build a runtime for each platform.
+
+```shell
+docker build . --quiet --platform=linux/amd64 --tag runtime-amd64
+sha256:fdf40374cc6f0b46191499fbc1dbbb05ddb76aca854f69f2912e580cfe624b4b
+```
+
+```shell
+docker build . --quiet --platform=linux/arm64 --tag runtime-arm64
+sha256:cb015ceabf46d2a55ccaeebb11db5659a2fb5e93de36713364efcf6d699069af
+```
+
+{{<hint "tip">}}
+You can use whatever tag you want. There's no need to push the runtime images to
+a registry. The tag is only used to tell `crossplane xpkg build` what runtime to
+embed.
+{{</hint>}}
+
+Use the Crossplane CLI to build a package for each platform. Each package embeds
+a runtime image. 
+
+The {{<hover label="build" line="2">}}--package-root{{</hover>}} flag specifies
+the `package` directory, which contains `crossplane.yaml`. This includes
+metadata about the package.
+
+The {{<hover label="build" line="3">}}--embed-runtime-image{{</hover>}} flag
+specifies the runtime image tag built using Docker.
+
+The {{<hover label="build" line="4">}}--package-file{{</hover>}} flag specifies
+specifies where to write the package file to disk. Crossplane package files use
+the extension `.xpkg`.
+
+```shell {label="build"}
+crossplane xpkg build \
+    --package-root=package \
+    --embed-runtime-image=runtime-amd64 \
+    --package-file=function-amd64.xpkg
+```
+
+```shell
+crossplane xpkg build \
+    --package-root=package \
+    --embed-runtime-image=runtime-arm64 \
+    --package-file=function-arm64.xpkg
+```
+
+{{<hint "tip">}}
+Crossplane packages are special OCI images. Read more about packages in the
+[packages documentation]{{<ref "../../master/concepts/packages">}}.
```

Comment:
this link is broken on the preview site and looks like this:
```
[packages documentation]https://deploy-preview-597–crossplane.netlify.app/master/concepts/packages/.
```

Reply:
Thanks! I always forget that you need three levels of braces on these - `({{<>}})`.